core/jobs/batch_jobs/opportunity_management_jobs.py

```diff
@@ -35,25 +37,190 @@
 from core.platform import models
 
 import apache_beam as beam
-
 from typing import Dict, List, Union
 
 MYPY = False
 if MYPY: # pragma: no cover
     from mypy_imports import exp_models
     from mypy_imports import opportunity_models
+    from mypy_imports import question_models
+    from mypy_imports import skill_models
     from mypy_imports import story_models
     from mypy_imports import topic_models
 
 (
     exp_models, opportunity_models, story_models,
-    topic_models
+    topic_models, skill_models, question_models
 ) = models.Registry.import_models([
     models.NAMES.exploration, models.NAMES.opportunity, models.NAMES.story,
-    models.NAMES.topic
+    models.NAMES.topic, models.NAMES.skill, models.NAMES.question
 ])
 
 
+class DeleteSkillOpportunityModelJob(base_jobs.JobBase):
+    """Job that deletes SkillOpportunityModels."""
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        """Returns a PCollection of 'SUCCESS' or 'FAILURE' results from
+        deleting SkillOpportunityModel.
+
+        Returns:
+            PCollection. A PCollection of 'SUCCESS' or 'FAILURE' results from
+            deleting SkillOpportunityModel.
+        """
+        skill_opportunity_model = (
+            self.pipeline
+            | 'Get all non-deleted skill models' >> ndb_io.GetModels(
+                opportunity_models.SkillOpportunityModel.get_all(
+                    include_deleted=False))
+        )
+
+        unused_delete_result = (
+            skill_opportunity_model
+            | beam.Map(lambda model: model.key)
+            | 'Delete all models' >> ndb_io.DeleteModels()
+        )
+
+        return (
+            skill_opportunity_model
+            | 'Count all new models' >> beam.combiners.Count.Globally()
+            | 'Only create result for new models when > 0' >> (
+                beam.Filter(lambda n: n > 0))
+            | 'Create result for new models' >> beam.Map(
+                lambda n: job_run_result.JobRunResult(stdout='SUCCESS %s' % n))
+        )
```

Comment:
Please use `job_result_transforms.CountObjectsToJobRunResult()` here, look at the latest `DeleteExplorationOpportunitySummariesJob` to see how it is used.

Reply:
Done, thanks!